torch/multiprocessing/reductions.py

```diff
@@ -23,10 +23,108 @@ def __init__(self, ptr):
         self.cdata = ptr
 
 
-# mapping from handles to StorageRef objects
+# mapping from handles to StorageRef objects. Doesn't include CUDA handles.
 shared_cache = weakref.WeakValueDictionary()
 
 
+# CUDA storage caching strategy
+# -----------------------------
+#
+# All CUDA storages are cached into 'cuda_cache'.
+#
+# Call a "CUDA storage that processes wish to share" a "real storage".
+# "real storages" are the storages that other processes want to and
+# eventually will receive.
+#
+# Call a "CUDA storage that points to the base of an allocation block" a
+# "base storage". These are obtained by opening CUDA handles to the
+# base of an allocation block, where multiple "real storages" may live.
+# Those "real storages" can be obtained by adding an offset to "base storage"
+#
+# "base storages" are cached into the "base cache" and "real storages" are
+# cached into the "real cache"
+#
+# When a process shares a cuda storage:
```

Comment:
`shares` is very unclear in this context. Since you use `receive` to indicate the other direction, `send` seems more appropriate.